ThePensionsRegulator.Frontend/wwwroot/tpr/tpr-search.js

```diff
@@ -0,0 +1,124 @@
+
+        const accordionHeader = createElementWithClassName("div", "govuk-accordion__section-header");
+
+        const accordionHeadingElement = createElementWithClassName("h3", "govuk-accordion__section-heading"); 
+
+        const accordionHeadingElementText = document.createTextNode(heading);
+
+        const accordionHeadingButtonElement = createElementWithClassName("span", "govuk-accordion__section-button");
+        accordionHeadingButtonElement.setAttribute("id", `accordion1-heading-${index}`);
+        accordionHeadingButtonElement.appendChild(accordionHeadingElementText);
+
+        accordionHeadingElement.appendChild(accordionHeadingButtonElement);
+        accordionHeader.appendChild(accordionHeadingElement);
+
+        const accordionContent = document.createElement("div");
+        accordionContent.className = "govuk-accordion__section-content";
+        accordionContent.innerHTML = content;
+
+        accordionSection.appendChild(accordionHeader);
+        accordionSection.appendChild(accordionContent);
+
+        return accordionSection;
+    }
+
+    async function fetchContentById(contentId) {
+        const response = await fetch(`${apiUrl}/api/faq/content/${contentId}`, { method: 'GET', headers: { 'Content-Type': 'application/json' } });
+        return await response.json(); 
+    }
+
+    async function buttonOnClick() {
+        const searchInput = document.getElementById("ask-input");
+        const searchValue = searchInput.value
+
+        const response = await fetch(`${apiUrl}/api/faq/content/search?searchTerm=${searchValue}&page=1&pageSize=5`, { method: 'GET', headers: { 'Content-Type': 'application/json' } });
+
+
+        const results = (await response.json()).results;
+
+        removeAccordion("search-results-accordion");
+
+        if (results.length === 0) {
+            const noResultsHeading = createNoResultsFoundHeading();
+            const searchBlock = document.getElementsByClassName("tpr-search-results__input")[0];
+            searchBlock.after(noResultsHeading);
+        } else {
+
+            const accordionSections = [];
+            await Promise.all(results.map(async (result, index) => {
+                const content = await fetchContentById(result.key);
+                const header = content.name;
+                const pageContent = content.pageContent;
+
+                accordionSections.push(createAccordionSection(header, pageContent, index))
+            }));
+            createAccordion(accordionSections);
+
+        }
+
+    }
+
+    function createElementWithClassName(elementName, className) {
+        const element = document.createElement(elementName);
+        element.className = className;
+
+        return element;
+    }
+
+    function createNoResultsFoundHeading() {
+        const noResultsHeading = document.createElement("h3");
+        noResultsHeading.className = "govuk-heading-m";
+        noResultsHeading.textContent = "No Results Found";
```

Comment:
Text should be sentence case... and you probably need it to be localisable too.

This shows how GOV.UK components handle localisation in JS: https://github.com/thepensionsregulator/govuk-frontend-aspnetcore-extensions/blob/develop/GovUk.Frontend.AspNetCore.Extensions/wwwroot/govuk/govuk-js-init.js